### PR TITLE
feat: add http-post support

### LINKED ISSUE #82: Add HTTP-POST support
Post request needs 2 new fields:
- headers
- body

### PR DESCRIPTION
**Pull request by Tommytrg**
close #82

### CODE REVIEW

**Comment by aesedepece** [comment]
Please take into account that HTTP-POST requests, in addition to the `headers` field, also have a `body` field (e.g. that's where you'd put a GraphQL query) :wink:

File: src/constants.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const DEFAULT_SCRIPT_FIRST_TYPE = OutputType.String
 export const KIND_OPTIONS = [Kind.HttpGet, Kind.RNG]
```
Comment (gabaldon) [inline src/constants.ts]: You should add `Kind.HttpPost` to KIND_OPTIONS
```suggestion
export const KIND_OPTIONS = [Kind.HttpGet, Kind.RNG, Kind.HttpPost]
```

**Review by gabaldon — CHANGES_REQUESTED** [review]
You need to add support for post sources in `radon.ts` constructor.
`this._sourceType = radRequest.retrieve.find((source) => source.kind === Kind.RNG)
      ? Kind.RNG
      : Kind.HttpGet`
This line unable updating the source type to Kind.HttpPost.

File: src/source.ts [hunk 2]
```diff
@@ -12,13 +12,16 @@ import { KIND_OPTIONS, CONTENT_TYPE_OPTIONS } from './constants'
 import { Cache } from './structures'
 import { Script } from './script'
 import { I18n } from './i18n'
+import { OutgoingHttpHeaders } from 'http'
```
Comment (gabaldon) [inline src/source.ts:15]: You need to add `body` and `headers` to the update function, otherwise it will not update.

**Pull request merged**